pkg/reduce/data_forward.go

```diff
@@ -464,12 +464,13 @@ messagesLoop:
 // writeToPBQ writes to the PBQ. It will return error only if it is not failing to write to PBQ and is in a continuous
 // error loop, and we have received ctx.Done() via SIGTERM.
 func (df *DataForward) writeToPBQ(ctx context.Context, m *isb.ReadMessage, p partition.ID, kw window.AlignedKeyedWindower) error {
-	startTime := time.Now()
-	defer pbqWriteTime.With(map[string]string{
-		metrics.LabelVertex:             df.vertexName,
-		metrics.LabelPipeline:           df.pipelineName,
-		metrics.LabelVertexReplicaIndex: strconv.Itoa(int(df.vertexReplica)),
-	}).Observe(float64(time.Since(startTime).Milliseconds()))
+	defer func(t time.Time) {
+		metrics.PBQWriteTime.With(map[string]string{
+			metrics.LabelVertex:             df.vertexName,
+			metrics.LabelPipeline:           df.pipelineName,
+			metrics.LabelVertexReplicaIndex: strconv.Itoa(int(df.vertexReplica)),
+		}).Observe(float64(time.Since(t).Milliseconds()))
+	}(time.Now())
```

Comment:
nice, I was under an impression that `time.Now()` is evaluated when defer function executes. It turns out it's evaluated immediately when encountering the defer statement. Learnt something new.